mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+import csv
+import json
+import os.path as osp
+import pickle
+import warnings
+from collections import OrderedDict, defaultdict
+
+import numpy as np
+import torch.distributed as dist
+from mmcv.runner import get_dist_info
+from mmcv.utils import print_log
+
+from mmdet.core import eval_map
+from .builder import DATASETS
+from .custom import CustomDataset
+
+
+@DATASETS.register_module()
+class OpenImagesDataset(CustomDataset):
+    """Open Images dataset for detection."""
+
+    def __init__(self,
+                 label_description_file='',
+                 image_level_ann_file='',
+                 get_supercategory=True,
+                 hierarchy_file=None,
+                 get_metas=True,
+                 load_from_file=True,
+                 meta_file='',
+                 filter_labels=True,
+                 load_image_level_labels=True,
+                 **kwargs):
+        """
+        Args:
+            label_description_file (str): File path to the label map proto.
```

Comment:
what is label map proto?

Reply:
The class names in MID format can be converted to their short descriptions by looking into class-descriptions.csv:
an example: 
```
/m/0pc9,Alphorn
/m/0pckp,Robin
/m/0pcm_,Larch
/m/0pcq81q,Soccer player
```
any name suggested?

Reply:
label_file (str): File path of the label description file that maps the classes names in MID format to their short descriptions.